File: sdk/php/src/events/RemoveFragments.php
<?php
/**
 * @copyright Copyright (c) PutYourLightsOn
 */

namespace aereaco\nexus-ux\events;

use aereaco\nexus-ux\Consts;
use aereaco\nexus-ux\enums\EventType;

class RemoveFragments implements EventInterface
{
    use EventTrait;

    public string $selector;
    public bool $useViewTransition = Consts::DEFAULT_FRAGMENTS_USE_VIEW_TRANSITIONS;

    public function __construct(string $selector, array $options = [])
    {
        $this->selector = $selector;

        foreach ($options as $key => $value) {
            $this->$key = $value;
        }
    }

    /**
     * @inerhitdoc
     */
    public function getEventType(): EventType
    {
        return EventType::RemoveFragments;
    }

    /**
     * @inerhitdoc
     */
    public function getDataLines(): array
    {
        $dataLines = [
            $this->getDataLine(Consts::SELECTOR_DATALINE_LITERAL, $this->selector),
        ];

        if ($this->useViewTransition !== Consts::DEFAULT_FRAGMENTS_USE_VIEW_TRANSITIONS) {
            $dataLines[] = $this->getDataLine(Consts::USE_VIEW_TRANSITION_DATALINE_LITERAL, $this->getBooleanAsString($this->useViewTransition));
        }

        return $dataLines;
    }
}

File: sdk/java/core/src/main/java/aereaco/nexusux/RemoveFragments.java
package aereaco.nexusux;

import aereaco.nexusux.enums.EventType;

import java.util.ArrayList;
import java.util.List;

public class RemoveFragments implements Event {
    private final String selector;
    private boolean useViewTransition = Consts.DEFAULT_FRAGMENTS_USE_VIEW_TRANSITIONS;
    private String eventId;
    private Integer retryDuration;

    public RemoveFragments(String selector) {
        this.selector = selector;
    }

    public RemoveFragments(String selector, boolean useViewTransition) {
        this.selector = selector;
        this.useViewTransition = useViewTransition;
    }

    @Override
    public EventType getEventType() {
        return EventType.RemoveFragments;
    }

    @Override
    public List<String> getDataLines() {
        List<String> dataLines = new ArrayList<>();
        dataLines.add(String.format("data: %s %s", Consts.SELECTOR_DATALINE_LITERAL, selector));
        if (useViewTransition != Consts.DEFAULT_FRAGMENTS_USE_VIEW_TRANSITIONS) {
            dataLines.add(String.format("data: %s %s", Consts.USE_VIEW_TRANSITION_DATALINE_LITERAL, useViewTransition));
        }
        return dataLines;
    }

    @Override
    public String getEventId() {
        return eventId;
    }

    @Override
    public Integer getRetryDuration() {
        return retryDuration;
    }
}
